src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -75,114 +80,115 @@
      * message in the "messages.properties" file.
      */
     public static final String CREATE_LISTENER_EXCEPTION = "Main.createListener";
+
+    /** Name for the moduleConfig attribute 'tabWidth'. */
+    private static final String ATTRIB_TAB_WIDTH_NAME = "tabWidth";
+
     /** Logger for Main. */
     private static final Log LOG = LogFactory.getLog(Main.class);
 
     /** Width of CLI help option. */
     private static final int HELP_WIDTH = 100;
 
+    /** Exit code returned when user specified invalid command line arguments. */
+    private static final int EXIT_WITH_INVALID_USER_INPUT_CODE = -1;
+
     /** Exit code returned when execution finishes with {@link CheckstyleException}. */
     private static final int EXIT_WITH_CHECKSTYLE_EXCEPTION_CODE = -2;
 
-    /** Name for the option 'v'. */
-    private static final String OPTION_V_NAME = "v";
-
-    /** Name for the option 'c'. */
-    private static final String OPTION_C_NAME = "c";
-
-    /** Name for the option 'f'. */
-    private static final String OPTION_F_NAME = "f";
-
-    /** Name for the option 'p'. */
-    private static final String OPTION_P_NAME = "p";
-
-    /** Name for the option 'o'. */
-    private static final String OPTION_O_NAME = "o";
-
-    /** Name for the option 's'. */
-    private static final String OPTION_S_NAME = "s";
-
-    /** Name for the option 't'. */
-    private static final String OPTION_T_NAME = "t";
-
-    /** Name for the option '--tree'. */
-    private static final String OPTION_TREE_NAME = "tree";
-
-    /** Name for the option 'tabWidth'. */
-    private static final String OPTION_TAB_WIDTH_NAME = "tabWidth";
+    /** The default number of threads to use for checker and the tree walker. */
+    private static final int DEFAULT_THREAD_COUNT = 1;
 
-    /** Name for the option '-T'. */
-    private static final String OPTION_CAPITAL_T_NAME = "T";
+    /** Default distance between tab stops. */
+    private static final int DEFAULT_TAB_WIDTH = 8;
 
-    /** Name for the option '--treeWithComments'. */
-    private static final String OPTION_TREE_COMMENT_NAME = "treeWithComments";
+    /** Default output format. */
+    private static final OutputFormat DEFAULT_OUTPUT_FORMAT = OutputFormat.plain;
 
-    /** Name for the option '-j'. */
-    private static final String OPTION_J_NAME = "j";
+    /** Config file location. */
+    @Option(names = "-c", description = "Sets the check configuration file to use.",
+            required = true)
```

Comment:
This is not always required.
Example: http://checkstyle.sourceforge.net/writingchecks.html#Tool_to_print_Java_tree_structure

Reply:
Thanks for pointing that out! I’ll remove the `required` attribute and put back the validation logic.